src/main/controllers/CopyToOtherPartyController.ts

```diff
@@ -0,0 +1,108 @@
+import { Response } from 'express';
+
+import { Form } from '../components/form/form';
+import { isFieldFilledIn } from '../components/form/validator';
+import { AppRequest } from '../definitions/appRequest';
+import { YesOrNo } from '../definitions/case';
+import { PageUrls, TranslationKeys } from '../definitions/constants';
+import { FormContent, FormFields } from '../definitions/form';
+import { AnyRecord } from '../definitions/util-types';
+import { getLogger } from '../logger';
+
+import { setUserCase, submitClaimantTse } from './helpers/CaseHelpers';
+import { getPageContent } from './helpers/FormHelpers';
+import { getLanguageParam } from './helpers/RouterHelpers';
+
+const logger = getLogger('CopyToOtherPartyController');
+
+export default class CopyToOtherPartyController {
+  private readonly form: Form;
+
+  private readonly CopyToOtherPartyContent: FormContent = {
+    fields: {
+      copyCorrespondence: {
+        classes: 'govuk-radios',
+        id: 'copyCorrespondence',
+        type: 'radios',
+        label: (l: AnyRecord): string => l.legend,
+        labelHidden: false,
+        labelSize: 'l',
+        values: [
+          {
+            name: 'copyCorrespondence',
+            label: (l: AnyRecord): string => l.yes,
+            value: YesOrNo.YES,
+          },
+          {
+            name: 'copyCorrespondence',
+            label: (l: AnyRecord): string => l.no,
+            value: YesOrNo.NO,
+            subFields: {
+              noCopyReason: {
+                id: 'noCopyReason',
+                name: 'noCopyReason',
+                type: 'textarea',
+                label: (l: AnyRecord): string => l.hintText,
+                labelSize: 'm',
+                isPageHeading: true,
+                classes: 'govuk-textarea',
+                attributes: { maxLength: 1500 },
+                validator: isFieldFilledIn,
+              },
+            },
+          },
+        ],
+        validator: isFieldFilledIn,
+      },
+    },
+    submit: {
+      text: (l: AnyRecord): string => l.continue,
+      classes: 'govuk-!-margin-right-2',
+    },
+  };
+
+  constructor() {
+    this.form = new Form(<FormFields>this.CopyToOtherPartyContent.fields);
+  }
+
+  public post = async (req: AppRequest, res: Response): Promise<void> => {
+    setUserCase(req, this.form);
+    // Moved submission to here temporarily
+    await submitClaimantTse(req, logger);
```

Comment:
When submitting a case, you need to reset claimantTse fields

Reply:
Added